src/ui/public/vislib/lib/axis.js

```diff
@@ -68,7 +70,7 @@ export default function AxisFactory(Private) {
     }
 
     getLength(el, n) {
-      const margin = this.config.get('vis._attr.margin');
+      const margin = this.config._chartConfig.get('style.margin');
```

Comment:
I don't think we should be accessing the `chartConfig` as a property of the `axisConfig`. Maybe the `axisConfig` could use the `chartConfig` to populate defaults and expose them as config values?

In the case of margin, it would look something like:
```js
class AxisConfig {
  constructor(chartConfig) {
    const defaults = defaultsDeep({}, standardDefaults, {
      style: chartConfig.get('style')
    }
  }
}
```

Then, users could define a `style.margin` for the axis, but it would default to the chart's `style.margin`. Thoughts?